### ISSUE TITLE
Kong custom plugin "1.x.x": kong.api.crud_helpers  not working

### ISSUE THREAD

**Issue by sharmmoh1983** [Kong/kong#4278]
I am trying to build one custom plugin which sends response using kong.tools.responses but it seems it is removed in new version of Kong 1.x.x

What is the alternative of using kong.tools.responses in plugins for new version.

```
2019/02/04 04:24:01 [error] 304#0: *1923 lua entry thread aborted: runtime error: /usr/local/share/lua/5.1/kong/tools/utils.lua:584: /usr/local/share/lua/5.1/kong/plugins/rbac/api.lua:3: module 'kong.api.crud_helpers' not found:No LuaRocks module found for kong.api.crud_helpers
	no field package.preload['kong.api.crud_helpers']
	no file './kong/api/crud_helpers.lua'
	no file './kong/api/crud_helpers/init.lua'
	no file '/usr/local/openresty/site/lualib/kong/api/crud_helpers.ljbc'
	no file '/usr/local/openresty/site/lualib/kong/api/crud_helpers/init.ljbc'
	no file '/usr/local/openresty/lualib/kong/api/crud_helpers.ljbc'
	no file '/usr/local/openresty/lualib/kong/api/crud_helpers/init.ljbc'
	no file '/usr/local/openresty/site/lualib/kong/api/crud_helpers.lua'
	no file '/usr/local/openresty/site/lualib/kong/api/crud_helpers/init.lua'
	no file '/usr/local/openresty/lualib/kong/api/crud_helpers.lua'
	no file '/usr/local/openresty/lualib/kong/api/crud_helpers/init.lua'
	no file '/usr/local/openresty/luajit/share/luajit-2.1.0-beta3/kong/api/crud_helpers.lua'
	no file '/usr/local/share/lua/5.1/kong/api/crud_helpers.lua'
	no file '/usr/local/share/lua/5.1/kong/api/crud_helpers/init.lua'
	no file '/usr/local/openresty/luajit/share/lua/5.1/kong/api/crud_helpers.lua'
	no file '/usr/local/openresty/luajit/share/lua/5.1/kong/api/crud_helpers/init.lua'
	no file '/home/kong/.luarocks/share/lua/5.1/kong/api/crud_helpers.lua'
	no file '/home/kong/.luarocks/share/lua/5.1/kong/api/crud_helpers/init.lua'
	no file '/usr/local/openresty/site/lualib/kong/api/crud_help
```

**Comment by jeremyjpj0916** [comment]
Dup of #4277 essentially, same answer: Changelog: https://github.com/Kong/kong/blob/master/CHANGELOG.md#plugins-3 to be specific for the crud helpers section 

"The kong.api.crud_helpers module was removed (deprecated since the introduction of the new DAO in 0.13.0). Use kong.api.endpoints instead if you need to customize the auto-generated endpoints."

**Comment by sharmmoh1983** [comment]
@jeremyjpj0916 
Thanks for your input . I have to know re engineer everything to make my existing code align with new changes

Is there any way to do auto transtion to new format especially the new Dao layer may be using  best-effort schema auto-translator

Please let me know the steps to do so or some documentation for the same

**Comment by thibaultcha** [comment]
@jeremyjpj0916 is also correct, 1.0 auto-generated Admin API endpoints when the underlying entities use our new DAO implementation, which is a huge improvement (the endpoints are also more robust and user-friendly regarding support content-types, inferences, and error messages).

You can find more details for these changes and the path forward in the [1.0 Upgrade Path](https://github.com/Kong/kong/blob/master/UPGRADE.md#plugins). Also keep in mind that our open source plugins are a good reference _and_ some of them are really simple to follow (e.g. key-auth, basic-auth...)

> I have to know re engineer everything to make my existing code align with new changes

We released 0.15 alongside 1.0 with support for all deprecated concepts (these Admin API utilities, the old "API" entity, etc...)  for users not willing to update their code. Maybe you should consider it until you can make the jump to 1.0?

Considering this answered.
